src/main/javadoc/overview.html

```diff
@@ -0,0 +1,362 @@
+        </div>
+    </div>
+</details>
+
+<details>
+    <summary>Advanced: State Machine Diagram</summary>
+
+    <h3>PIC Internal Stream States</h3>
+    <p>The native PIC library manages these internal states. This is the expected &quot;normal&quot; state transition path for the Stream.</p>
+    <div style="background-color: #f5f5f5; padding: 20px; margin: 20px 0; border-radius: 5px;">
+        <div style="text-align: center; margin: 20px 0;">
+            <span style="display: inline-block; padding: 8px 12px; margin: 3px; background-color: #fff3e0; border: 2px solid #ff9800; border-radius: 5px; text-align: center; min-width: 100px;">NEW</span>
+            <span style="margin: 0 8px; font-size: 16px; color: #2196f3;">&rarr;</span>
+            <span style="display: inline-block; padding: 8px 12px; margin: 3px; background-color: #e8f5e8; border: 2px solid #4caf50; border-radius: 5px; text-align: center; min-width: 100px;">DESCRIBE</span>
+            <span style="margin: 0 8px; font-size: 16px; color: #2196f3;">&rarr;</span>
+            <span style="display: inline-block; padding: 8px 12px; margin: 3px; background-color: #e8f5e8; border: 2px solid #4caf50; border-radius: 5px; text-align: center; min-width: 100px;">CREATE</span>
+            <span style="margin: 0 8px; font-size: 16px; color: #2196f3;">&rarr;</span>
+            <span style="display: inline-block; padding: 8px 12px; margin: 3px; background-color: #e8f5e8; border: 2px solid #4caf50; border-radius: 5px; text-align: center; min-width: 100px;">GET_ENDPOINT</span>
+            <span style="margin: 0 8px; font-size: 16px; color: #2196f3;">&rarr;</span>
+            <span style="display: inline-block; padding: 8px 12px; margin: 3px; background-color: #e8f5e8; border: 2px solid #4caf50; border-radius: 5px; text-align: center; min-width: 100px;">GET_TOKEN</span>
+            <span style="margin: 0 8px; font-size: 16px; color: #2196f3;">&rarr;</span>
+            <span style="display: inline-block; padding: 8px 12px; margin: 3px; background-color: #fff3e0; border: 2px solid #ff9800; border-radius: 5px; text-align: center; min-width: 100px;">READY</span>
+            <span style="margin: 0 8px; font-size: 16px; color: #2196f3;">&rarr;</span>
+            <span style="display: inline-block; padding: 8px 12px; margin: 3px; background-color: #e8f5e8; border: 2px solid #4caf50; border-radius: 5px; text-align: center; min-width: 100px;">PUT_STREAM</span>
+            <span style="margin: 0 8px; font-size: 16px; color: #2196f3;">&rarr;</span>
+            <span style="display: inline-block; padding: 8px 12px; margin: 3px; background-color: #e8f5e8; border: 2px solid #4caf50; border-radius: 5px; text-align: center; min-width: 100px;">STREAMING</span>
+        </div>
+    </div>
+
+</details>
+
+<details>
+    <summary>Advanced: JNI flow &amp; Deep Dive</summary>
```

Comment:
Might be a good place somewhere in this doc for a brief introduction to what a JNI is incase a KVS user is new to it.